src/connect.ts

```diff
@@ -73,11 +80,15 @@ async function connectWithFailFast(uri: string, client: MongoClient, logger: Con
   try {
     await client.connect();
   } catch (err: unknown) {
+    let connectErr = err;
     if (failEarlyClosePromise !== null) {
       await failEarlyClosePromise;
-      throw failedConnections.values().next().value; // Just use the first failure.
+      connectErr = failedConnections.values().next().value; // Just use the first failure.
+    }
+    if ((connectErr as any)?.constructor?.name === 'MongoServerSelectionError' && isAtlas(uri)) {
+      (connectErr as Error).message = `${(connectErr as Error).message}. It looks like this is a MongoDB Atlas cluster. Please ensure that your Network Access List allows connections from your IP.`;
```

Comment:
[not blocking]

I think there might still be some confusion here around the `catch (err: unknown)` vs `catch (err: any)` …

`err: unknown` is the best match when we’re expecting an error being thrown from code outside of our control, which might realistically throw values that aren’t `Error` instances, where we want to force ourselves to carefully do typechecking in TS with `typeof err` and `'...' in err`.

`err: any` is reasonable when we can assume that `err` is always an `Error` object, and we don’t really need to do manual typechecking in the places where we access it.

But here we’re mixing the two, we’re using `err: unknown` but then later basically assuming that `err` is an `Error` object anyway. That’s okay, but you could as well just say `err: any` in the first place then.